src/platform/backends/qemu/qemu_virtual_machine.cpp

```diff
@@ -244,6 +244,18 @@ mp::QemuVirtualMachine::QemuVirtualMachine(const VirtualMachineDescription& desc
             vm_process->write(QJsonDocument(qmp).toJson());
         },
         Qt::QueuedConnection);
+
+    QObject::connect(
+        this,
+        &QemuVirtualMachine::on_synchronize_clock,
+        this,
+        [this](const OpenSSHKeyProvider* key_provider) {
```

Comment:
The reason I didn't use `SSHKeyProvider` is because it's a abstract class and Qt has a hard time serializing abstract classes to pass through signal/slots without some special modifications to the class that I wasn't able to figure out. Of the two concrete child classes, `OpenSSHKeyProvider` should be the only one that will get passed.

Reply:
I've been thinking about this some more and I really don't think this needs to be in a signal/slot.  I think you can instead just make the call directly to set the instance clock in `wait_until_ssh_up()`.  Other than it may take a very slight amount of time to make the call, I don't really see any downside to doing it this way.

As an aside, part of the reason the const ref isn't working is because it's a `QueuedConnection` and the variable that it refers to goes out of scope before the lambda is called.

Reply:
I guess we could the ssh call in `wait_until_ssh_up()`, but it's nice having this done asynchronously (which is how I understand `QueuedConnection` to work). I went back though and took a another look at this and got it to work using `SSHKeyProvider`.

Reply:
Much better now, thanks!